### DISCUSSION TITLE
Astro.build - Help for proper setup of Tailwind.

### DISCUSSION THREAD

**Discussion by mrjohnmac** [tailwindlabs/tailwindcss#9903] [Help] [answered]
Hey people, 

I‘m starting out learning Astro. And I would like to build this on TailwindCSS. Astro supports Tailwind. But obviously, setting up Tailwind in general is not exactly a slick and simple process. It‘s just too confusing with the various areas of the config files setup. 

I have spent enough time now trying to figure this out and diving into threads and docs. 

Here are a few things I need to set up. 

- Flexible spacing using Clamp. Especially for side padding margins. 
- Flexible fonts sizes using Clamp. Not fixed sizes. 
- Define a container max-width for centred content. But also have a full-width container. 
- font family. 

I just can‘t figure out and land on the proper way to set up a proper "theming" of an dev. site while building on Astro. 

Let me know what other details I need to share.

**Comment by mrjohnmac** [comment]
Seems like not many people are using Astro? But I would think the setup would be similar to VueJS or React?

**Comment by gianlucajahn** [comment, marked as answer]
I can help with the first two things.

```
// Import Tailwind CSS and Astro
import "tailwindcss/dist/tailwind.min.css";
import { Astro, Clamp } from "astro-react";

function MyComponent() {
  return (
    <Astro>
      {/* Use Clamp to get flexible spacing */}
      <div className="p-4 md:p-8 lg:p-12 xl:p-16">
        <Clamp min="16" max="24" className="text-xl md:text-2xl lg:text-3xl xl:text-4xl">
          Some text with flexible font size
        </Clamp>
      </div>
    </Astro>
  );
}
```

In this example, we import Tailwind CSS and Astro at the top of the file. In our component, we use the <Astro> component to wrap our content, and then use the <Clamp> component to get flexible spacing and font sizes. We use the min and max props of the <Clamp> component to set the minimum and maximum values for the spacing and font size. We also use Tailwind CSS classes to set the initial values for the spacing and font size, and the <Clamp> component will adjust these values based on the minimum and maximum values we set. This will allow us to get flexible spacing and font sizes that are responsive to different screen sizes.

**Reply by mrjohnmac** [reply]
Hey there! Awesome, thank you for the feedback. This was a different approach than what I have been getting from the Astro and Tailwind Discord groups. But this one was also interesting, and I will save it to my snippets. Seems simpler to use at times. 

I did get the site fixed by suggested from the Tailwind group ✔️
